easybuild/tools/run.py

```diff
@@ -589,3 +589,59 @@ def parse_log_for_error(txt, regExp=None, stdout=True, msg=None):
                   (regExp, '\n'.join([x[0] for x in res])))
 
     return res
+
+
+def check_log_for_errors(logTxt, regExps):
+    """
+    Check logTxt for messages matching regExps in order and do appropriate action
+    :param logTxt: String containing the log, will be split into individual lines
+    :param regExps: List of: regular expressions (as RE or string) to error on,
+                    or tuple of regular expression and action (any of [IGNORE, WARN, ERROR])
+    """
+    global errors_found_in_log
+
+    def is_regexp_object(objToTest):
+        try:
+            objToTest.match('')
+            return True
+        except AttributeError:
+            return False
+
+    # Avoid accidentally passing a single element
+    assert isinstance(regExps, list), "regExps must be a list"
+    regExpTuples = []
+    for cur in regExps:
+        try:
+            if isinstance(cur, str):
+                regExpTuples.append((re.compile(cur), ERROR))
+            elif is_regexp_object(cur):
+                regExpTuples.append((cur, ERROR))
+            elif len(cur) != 2:
+                raise TypeError("Invalid tuple")
+            elif not isinstance(cur[0], str) and not is_regexp_object(cur[0]):
+                raise TypeError("Invalid RegExp in tuple")
+            elif cur[1] not in (IGNORE, WARN, ERROR):
+                raise TypeError("Invalid action in tuple")
+            else:
+                regExpTuples.append((re.compile(cur[0]) if isinstance(cur[0], str) else cur[0], cur[1]))
+        except TypeError:
+            raise EasyBuildError("Invalid input: No RegExp or tuple of RegExp and action: %s" % str(cur))
+    warnings = []
+    errors = []
+    for l in logTxt.split('\n'):
+        for regExp, action in regExpTuples:
+            m = regExp.search(l)
+            if m:
+                if action == ERROR:
+                    errors.append(l)
+                elif action == WARN:
+                    warnings.append(l)
+                break
+    errors_found_in_log += len(warnings) + len(errors)
+    if warnings:
+        _log.warning("Found %s potential errors in command output (output: %s)" %
+                     (len(warnings), ", ".join(warnings)))
+    if errors:
+        raise EasyBuildError("Found %s errors in command output (output: %s)" %
```

Comment:
`output:` should be `errors:`?

`errors` should be `error(s)` (since there could be only one, see your last test case)

Reply:
It was like that before and actually it is the output containing the error message. I'd just leave the name

Reply:
Oh and what'd you say about separating warning/error output lines by `\n` instead? Currently it is hard to read:
`== 2019-12-11 13:14:12,211 run.py:546 WARNING Found 26 errors in command output (output: /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/document.h:1881: warning: preprocessing issue while doing constant expression evaluation: syntax error, Searching for files in directory /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error, Preprocessing /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error/en.h..., Parsing file /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error/en.h..., Preprocessing /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error/error.h..., Parsing file /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error/error.h..., Parsing file /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/document.h:1881: warning: preprocessing issue while doing constant expression evaluation: syntax error, /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error/error.h:123: warning: Member operator==(const ParseResult &that) const  (function) of class rapidjson::ParseResult is not documented., /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error/error.h:124: warning: Member operator==(ParseErrorCode code) const  (function) of class rapidjson::ParseResult is not documented., /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error/error.h:125: warning: Member operator==(ParseErrorCode code, const ParseResult &err) (friend) of class rapidjson::ParseResult is not documented., Generating code for file include/rapidjson/error/en.h..., Generating code for file include/rapidjson/error/error.h..., Generating docs for file include/rapidjson/error/error.h..., /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/document.h:1881: warning: preprocessing issue while doing constant expression evaluation: syntax error, Searching for files in directory /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error, Preprocessing /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error/en.h..., Parsing file /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error/en.h..., Preprocessing /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error/error.h..., Parsing file /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error/error.h..., Parsing file /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1/tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/document.h:1881: warning: preprocessing issue while doing constant expression evaluation: syntax error, /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error/error.h:123: warning: Member operator==(const ParseResult &that) const  (function) of class rapidjson::ParseResult is not documented., /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error/error.h:124: warning: Member operator==(ParseErrorCode code) const  (function) of class rapidjson::ParseResult is not documented., /tmp/s3248973-EasyBuild/RapidJSON/1.1.0/GCCcore-8.2.0/rapidjson-1.1.0/include/rapidjson/error/error.h:125: warning: Member operator==(ParseErrorCode code, const ParseResult &err) (friend) of class rapidjson::ParseResult is not documented., Generating code for file include/rapidjson/error/en.h..., Generating code for file include/rapidjson/error/error.h..., Generating docs for file include/rapidjson/error/error.h...)
`

Reply:
So basically: `raise EasyBuildError("Found %s errors in command output. Output:\n\t%s)", ..., "\n\t".join(...))`

Reply:
one error per line makes sense, that's indeed a lot easier to read...

Reply:
Done